### PR TITLE
Various Utilities in Support of the Token-SDK

### PR DESCRIPTION
**Pull request by alexandrosfilios**
* Introduced mock context
* Optionally disable TLS option for FSC-node metrics, and added metrics for libp2p
* Added support for execution of views with web sockets (instead of HTTP requests). Support for remote wallets
* Client and server utilities for web sockets.

### CODE REVIEW

File: integration/fsc/pingpong/cmd/initiator/main.go [hunk 1]
```diff
@@ -19,6 +20,12 @@ func main() {
 		if err := registry.RegisterFactory("init", &pingpong.InitiatorViewFactory{}); err != nil {
 			return err
 		}
+		if err := registry.RegisterFactory("mockInit", &mock.InitiatorViewFactory{}); err != nil {
```
Comment (adecaro) [inline integration/fsc/pingpong/cmd/initiator/main.go:23]: why do we need this mockInit?

File: integration/fsc/pingpong/topology.go [hunk 2]
```diff
@@ -14,6 +14,7 @@ import (
 func Topology() []api.Topology {
 	// Create an empty FSC topology
 	topology := fsc.NewTopology()
+	topology.SetLogging("debug", "")
```
Comment (adecaro) [inline integration/fsc/pingpong/topology.go:17]: remove this

File: platform/view/sdk/sdk.go [hunk 3]
```diff
@@ -452,10 +454,17 @@ func (p *SDK) initWebOperationEndpointsAndMetrics() error {
 		}
 	}
 
+	prometheusTls := true
+	if configProvider.IsSet("fsc.metrics.prometheus.tls") {
+		prometheusTls = configProvider.GetBool("fsc.metrics.prometheus.tls")
+	}
+	logger.Infof("Starting prometheus with TLS: %v", prometheusTls)
```
Comment (adecaro) [inline platform/view/sdk/sdk.go:461]: but what if the provider is not prometheus?

File: platform/view/services/view/context.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+type Responders struct {
+	InitiatorView view.View
+	ResponderID   view.Identity
+	ResponderView view.View
+	Lock          sync.RWMutex
+	Channel       *session.LocalBidirectionalChannel
+}
+
+type MockContext struct {
```
Comment (adecaro) [inline platform/view/services/view/context.go:29]: why do we need this?

File: platform/view/services/view/stream.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+func GetStream(sp view.ServiceProvider) Stream {
+	scsBoxed, err := GetStreamIfExists(sp)
+	if err != nil {
+		panic(err)
+	}
+	return scsBoxed
+}
+func GetStreamIfExists(sp view.ServiceProvider) (Stream, error) {
```
Comment (adecaro) [inline platform/view/services/view/stream.go:27]: add a break line before

**Review by adecaro — CHANGES_REQUESTED** [review]

File: platform/view/sdk/sdk.go (hunk 3 above)
Comment (alexandrosfilios) [reply platform/view/sdk/sdk.go:461]: then the prometheus section in the config will not be defined, and the TLS option will default to false, as is the case right now that the variable doesn't exist.

File: integration/fsc/pingpong/mock/initiator.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+type Initiator struct {
+	*Params
+}
+
+func (p *Initiator) Call(ctx view.Context) (interface{}, error) {
+	// Retrieve responder identity
+	responder := view2.GetIdentityProvider(ctx).Identity("responder")
+	var context view.Context
+	if p.Mock {
+		c := &view3.MockContext{Ctx: ctx}
```
[the remark this answers is not in the document]
Comment (alexandrosfilios) [reply integration/fsc/pingpong/mock/initiator.go:33]: leaving as discussed

File: integration/fsc/pingpong/cmd/initiator/main.go (hunk 1 above)
Comment (alexandrosfilios) [reply integration/fsc/pingpong/cmd/initiator/main.go:23]: leaving as discussed

File: platform/view/services/view/context.go (hunk 4 above)
Comment (alexandrosfilios) [reply platform/view/services/view/context.go:29]: leaving as discussed

File: platform/view/sdk/sdk.go (hunk 3 above)
Comment (adecaro) [reply platform/view/sdk/sdk.go:461]: but it will still log `"Starting prometheus with TLS`.

Comment (alexandrosfilios) [reply platform/view/sdk/sdk.go:461]: oh indeed let me fix it

**Review by adecaro — APPROVED** [review]

**Pull request merged**